students/alexander_boone/lesson07/html_render.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    tag = "html"
+    indent = "    "
+
+    def __init__(self, content=None, **kwargs):
+        self.attributes = dict()
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+        for key, value in kwargs.items():
+            self.attributes[key] = value
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+    def render(self, out_file, cur_ind=""):
+        out_file.write(cur_ind)
+        out_file.write("<{}".format(self.tag))
+        if len(self.attributes.items()) > 0:
```

Comment:
this check is not needed, when you loop over a dictionary and it's empty it just wont' do anything which is the same effect